tests/debugging/test_rc.py

```diff
@@ -0,0 +1,189 @@
+                                            {
+                                                "language": "java",
+                                                "type": "snapshot",
+                                                "id": "probe3",
+                                                "active": True,
+                                                "where": {
+                                                    "typeName": "java.lang.String",
+                                                    "methodName": "concat",
+                                                    "signature": "String (String)",
+                                                },
+                                                "capture": {
+                                                    "maxReferenceDepth": 1,
+                                                    "maxCollectionSize": 100,
+                                                    "maxLength": 255,
+                                                    "maxFieldDepth": -1,
+                                                },
+                                                "sampling": {"snapshotsPerSecond": 1.0},
+                                                "tagMap": {},
+                                            },
+                                            {
+                                                "language": "java",
+                                                "id": "probe4",
+                                                "type": "snapshot",
+                                                "active": True,
+                                                "where": {
+                                                    "typeName": "java.lang.String",
+                                                    "methodName": "toUpperCase",
+                                                    "signature": "String ()",
+                                                },
+                                                "capture": {
+                                                    "maxReferenceDepth": 1,
+                                                    "maxCollectionSize": 100,
+                                                    "maxLength": 255,
+                                                    "maxFieldDepth": -1,
+                                                },
+                                                "sampling": {"snapshotsPerSecond": 1.0},
+                                                "tagMap": {},
+                                            },
+                                        ],
+                                    }
+                                ).encode()
+                            ).decode()
+                        },
+                    ]
+                }
+            ),
+        )
+
+        rc = DebuggingRC("uwsgi")
+
+        probes = wait_for(rc.get_probes, timeout=5, interval=0.5, cond=lambda _: _ is not None)
```

Comment:
Is it necessary to use `wait_for` because get_probes can hang? Or something else?

Reply:
Ah good point! I think this comes from tests against an actual service (i.e. not mocked), for which we needed to wait until the data propagated. With the mock tests here we don't need to wait! I'll clean this up.